pmdarima/arima/tests/test_arima.py

```diff
@@ -24,6 +24,25 @@
 import time
 import os
 
+# test images directories
+travis = os.environ.get("TESTING_ON_TRAVIS", "false").lower() == "true"
+
+if not travis:
+
+    # Only do this setup if we are not running on Travis CI.
+    from matplotlib.pyplot import savefig
+    from matplotlib.testing.exceptions import ImageComparisonFailure
+    import matplotlib._png as _png
+
+    test_images_input_path = \
+        os.path.join(os.path.dirname(__file__), 'input_images')
```

Comment:
This is why you're failing on Appveyor. The package is installed to site-packages, and this is going to look for the input_images path in site-packages, which fails. It's OK (and actually probably better) to use relative paths for something like this

Reply:
So I'm not a big fan of the relative path solution to look for the input images because you have to run from a certain path in order for the tests to work and pull in the images correctly and that path has to be the same for each CI environment. That would mean running AppVeyor and Circle CI would either both have to be executed from `empty_folder` since AppVeyor is using that or some commonly defined location. If I wrote the tests to look at that from a path in AppVeyor it would not work in CircleCI because the relative path would be different for that reason. It would mean mirroring work across all CI solutions in order for execution to be the same.

So as far as `site-packages` is concerned, I verified that is the case with both the Circle and AppVeyor installations that it is put in `site-packages`,  but the tests did execute and pass for Circle and not for AppVeyor, which to me was a difference in the setup/installation process as to why one would have visibility and not the other. So I decided to dig further and I think I have an exact answer :smiley:.

To fully understand the nature of this error, I tested pip-installing on the built artifacts, `pip install --pre --no-index --find-links dist/ pmdarima`,   and pip-installing using `pip install --editable pmdarima`. I believe the root cause for why it fails singly within AppVeyor as opposed to CircleCI as well is we are using the artifacts within the `dist/` directory to do the install of pmdarima for AppVeyor, and within Circle we are doing `pip install --editable pmdarima` which allows us to see the input images because the artifacts and the input images are colocated.

See excerpt from **appveyor.yml**:

```YAML

  # 9/25/18 - Use the scikit approach

  # Build the package (and the wheels!)

  - "%CMD_IN_ENV% pip install --timeout=60 -r build_tools/appveyor/requirements.txt"

  - "%CMD_IN_ENV% python setup.py bdist_wheel bdist_wininst" # <---- <<THIS LINE>>
  - ps: "ls dist"



  # Install the generated wheel file

  - "pip install --pre --no-index --find-links dist/ pmdarima" # <---- <<THIS LINE>>




# Not a .NET project, we build pmdarima in the install step instead

build: false



test_script:

  # Change to a non-source folder to make sure we run the tests on the

  # installed library

  - mkdir "../empty_folder"

  - "cd ../empty_folder"



  # Test with pytest

  - "pytest --showlocals --durations=20 --pyargs pmdarima"

  - "cd ../pmdarima"
```

That said, we should just be able to switch `pip install --pre --no-index --find-links dist/ pmdarima` with `pip install --editable pmdarima` and it should work. I verified this behavior locally to replicate the error and the workaround using the proposed solution and the artifacts are still built so this shouldn't affect the CD portion. Not sure if there is any reason why we can't point at the installed code directly for the tests so we have visibility to the image files. 

@tgsmith61591 Thoughts?